solara/components/file_input.py

```diff
@@ -0,0 +1,96 @@
+"""
+FileInput component.
+"""
+
+
+import threading
+from typing import Callable, List, Optional, cast
+
+from ipyvuetify.extra import FileInput as ExtraFileInput
+
+import solara
+import solara.hooks as hooks
+from solara.components.file_drop import FileInfo
+
+
+@solara.component
+def FileInput(
+    label: str = "",
+    on_total_progress: Optional[Callable[[float], None]] = None,
+    on_file: Optional[Callable[[List[FileInfo]], None]] = None,
```

Comment:
```suggestion
    on_file: Union[None, Callable[[FileInfo], None], Callable[[List[FileInfo]], None]] = None,
```
lets make the case for multiple=False simpler